src/Psalm/Config.php

```diff
@@ -2053,8 +2054,10 @@ private function getPHPVersionFromComposerJson(): ?string
             $php_version = $composer_json['require']['php'] ?? null;
 
             if (\is_string($php_version)) {
+                $version_parser = new VersionParser();
+                $min_php_version = $version_parser->parseConstraints($php_version)->getLowerBound()->getVersion();
                 foreach (['5.4', '5.5', '5.6', '7.0', '7.1', '7.2', '7.3', '7.4', '8.0'] as $candidate) {
-                    if (Semver::satisfies($candidate, $php_version)) {
+                    if (str_starts_with($min_php_version, $candidate)) {
```

Comment:
Can someone do a sanity check on this? I'm not 100% sure str_starts_with is correct in all cases.

Reply:
It's PHP 8 only to begin with :)

Reply:
I'm pretty sure psalm is using symfony's PHP 8 polyfill which includes this. It's working fine for me on PHP 7.3.

Reply:
It looks like the phar is failing on circle-ci though, so maybe that does need changed? All the unit tests are working fine though, so maybe the phar isn't being built correctly to include the polyfill?

Reply:
well, it's using symfony/console which require symfony/polyfill-php80 so it's an indirect dependancy. Maybe add the real dependancy?

Reply:
`symfony/console` is a non-dev dependency, which depends on symfony/polyfill-php80, so it seems like this should work and there's something wrong with the phar. Granted, if the polyfill is depended on it should be added as a direct dependency too, there's no guarantee `symfony/console` won't change.

Reply:
Hmm, doesn't seem to have worked. I'm happy to change it if we don't want the additional dependency, but I would like to know why it's not working.

Reply:
> Can someone do a sanity check on this? I'm not 100% sure `str_starts_with` is correct in all cases.

It'll select wrong version for two-digit minors: https://3v4l.org/jDb9B
I don't think PHP ever had two-digit minor, but theoretically it could happen.

Reply:
Is there any reason you got rid of `Semver::satisfies()`?

Reply:
https://github.com/vimeo/psalm/issues/5415#issuecomment-801281452 would be one I guess

Reply:
I think changing the call to `Semver::satisfies($min_php_version, "~$candidate.0")` would fix the issue: e.g. `7.2.5` would satisfy `~7.2.0`, as expected.

Reply:
> [#5415 (comment)](https://github.com/vimeo/psalm/issues/5415#issuecomment-801281452) would be one I guess

Yep, that.

Reply:
> I think changing the call to `Semver::satisfies($min_php_version, "~$candidate.0")` would fix the issue: e.g. `7.2.5` would satisfy `~7.2.0`, as expected.

Thanks, I think that'll work! I was stuck thinking about tweaking `$min_php_version`, I didn't even consider changing `$candidate`.